inc/file.class.php

```diff
@@ -299,119 +274,13 @@ public function post_updateItem($history = 1) {
       $itemtype = $this->getType();
       $itemId = $this->getID();
```

Comment:
These 2 lines are in File and Package, then you can move them in  the common implementation of publishTaskAction() (which I suggested to rename into post_updateItem())

Reply:
These lines aren't on the Package class.